lib/sisimai/bite/email/imailserver.rb

```diff
@@ -0,0 +1,181 @@
+module Sisimai::Bite::Email
+  module IMailServer
+    # Sisimai::Bite::Email::IMailServer parses a bounce email which created by
+    # Ipswitch IMail Server.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite::Email/IMailServer.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'x-mailer' => %r/\A[<]SMTP32 v[\d.]+[>][ ]*\z/,
+        :'subject'  => %r/\AUndeliverable Mail[ ]*\z/,
+      }
+      Re1 = {
+        :begin  => %r/\A\z/,    # Blank line
+        :error  => %r/Body of message generated response:/,
+        :rfc822 => %r/\AOriginal message follows[.]\z/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReSMTP = {
+        conn: %r{(?:
+             SMTP[ ]connection[ ]failed,
+            |Unexpected[ ]connection[ ]response[ ]from[ ]server:
+            )
+        },
+        ehlo: %r|Unexpected response to EHLO/HELO:|,
+        mail: %r|Server response to MAIL FROM:|,
+        rcpt: %r|Additional RCPT TO generated following response:|,
+        data: %r|DATA command generated response:|,
+      }
+      ReFailure = {
+        hostunknown: %r/Unknown[ ]host/x,
+        userunknown: %r/\A(?:Unknown[ ]user|Invalid[ ]final[ ]delivery[ ]userid)/x,
+        mailboxfull: %r/\AUser[ ]mailbox[ ]exceeds[ ]allowed[ ]size/x,
+        securityerr: %r/\ARequested[ ]action[ ]not[ ]taken:[ ]virus[ ]detected/x,
+        undefined:   %r/\Aundeliverable[ ]to[ ]/x,
+        expired:     %r/\ADelivery[ ]failed[ ]\d+[ ]attempts/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'IPSWITCH IMail Server'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Mailer']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from IMailServer
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
```

Comment:
__[rubocop]__
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->